integrated_channels/sap_success_factors/utils.py

```diff
@@ -0,0 +1,96 @@
+"""
+Utilities for Enterprise Integrated Channel SAP SuccessFactors.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import datetime
+import json
+
+from django.apps import apps
+
+from enterprise.lms_api import parse_lms_api_datetime
+from integrated_channels.integrated_channel.course_metadata import BaseCourseExporter
+
+
+UNIX_EPOCH = datetime.datetime(1970, 1, 1)
+UNIX_MIN_DATE_STRING = '1970-01-01T00:00:00Z'
+UNIX_MAX_DATE_STRING = '2038-01-19T03:14:07Z'
+
+
+class SapCourseExporter(BaseCourseExporter):
+    """
+    Class to provide data transforms for SAP SuccessFactors course export task.
+    """
+
+    def __init__(self, user, plugin_configuration):
+        super(SapCourseExporter, self).__init__(user, plugin_configuration)
+
+    def get_serialized_data(self):
+        final_structure = {
+            'ocnCourses': self.courses
+        }
+        return json.dumps(final_structure, sort_keys=True).encode('utf-8')
+
+    data_transform = {
+        'courseID': lambda x: x['key'],
+        'providerID': lambda x: apps.get_model(
+            'sap_success_factors',
+            'SAPSuccessFactorsGlobalConfiguration'
+        ).current().provider_id,
+        'status': lambda x: 'ACTIVE' if x['availability'] == 'Current' else 'INACTIVE',
+        'title': lambda x: [
+            {
+                'locale': 'English',
+                'value': x['title']
+            },
+        ],
+        'description': lambda x: [
+            {
+                'locale': 'English',
+                'value': x['full_description'] or '',
+            },
+        ],
+        'thumbnailURI': lambda x: (x['image']['src'] or ''),
+        'content': lambda x: [
```

Comment:
My understanding from the SAP docs was that this was supposed to be a list of sections in the course, with details about each section.

However, since this data isn't available from the Course Catalog, it's probably out of scope for this task.  It is available from the [Course Blocks API](http://edx.readthedocs.io/projects/edx-platform-api/en/latest/courses/blocks.html#example-response-showing-a-list-of-course-blocks-in-a-specified-course) though, if we need to have a follow-up task?